### PR TITLE
Better loops over alignment modes

### PR DESCRIPTION
**Pull request by cyrilbecot**
In EUTelGBL::init function, I have re-written the loop on alignment modes using actual loops to make it more compact. Then it might not be as "visual" but from a coding perspective it is a bit clearer and easier to develop around

### CODE REVIEW

File: processors/src/EUTelGBL.cc [hunk 1]
```diff
@@ -265,112 +270,36 @@ void EUTelGBL::init() {
...
-		      }
-		      if(std::find(_FixedZRot.begin(), _FixedZRot.end(), _sensorIDVec[ipl]) == _FixedZRot.end()) {
-                steerFile << (_sensorIDVec[ipl] * 10 + 6) << "  0.0  0.0" << endl; // LABELLING TO BE CHECKED
-	          } else {
-                steerFile << (_sensorIDVec[ipl] * 10 + 6) << "  0.0  -1.0" << endl; 
-		      }
+            if(_alignMode == Utility::alignMode::XYShiftsRotZ || _alignMode == Utility::alignMode::XYZShiftsRotZ || _alignMode == Utility::alignMode::XYZShiftsRotXYZ) alignModeArray.push_back(_FixedZRot);
+            if(alignModeArray.size() == 2 && _alignMode != Utility::alignMode::XYShifts ) continue;
+            
+            int id = 1;
```
Comment (edorossi) [inline processors/src/EUTelGBL.cc]: Can you change this variable name? Right now it is easy to confuse it to the sensor ID, at a first glance :)

**Comment by edorossi** [comment]
This looks very nice!! There is a small problem with the mode XYZShiftsRotZ, though. Because of the order definition later in the code, the zrotation corresponds to 3, and the zshift to 4, while you defined them the other way around. Note that in XYZShiftsRotXYZ they correspond to 6 and 3, respectively, thus with a different order... 
Of course, we should probably fix the rest of the code to be more consistent

**Comment by cyrilbecot** [comment]
Ok, the last two commits should fix that numbering issue. But yes it'd be more elegant to make it consistent throughout the code... Could you please check that I got it correctly for now ? We may fix the overall coherence issue later it's not very important for now imho

**Comment by edorossi** [comment]
Hi Cyril,

I agree, for now it should just work as intended, the style is for later. I will take a look later and let you know, but I am pretty sure that now it is fine!

Cheers,

E.

**Comment by edorossi** [comment]
Last thing: can you rename the variable “id” in the last loop? I know that I am nitpicking, but you used the same name a few lines before for the sensorID, and in this way it may be a bit confusing and bug-probe

**Comment by cyrilbecot** [comment]
Hey,
I've renamed it to "counter". Once the CI jobs are over we can merge i guess ?

**Review by edorossi — APPROVED** [review]

**Pull request merged**